src/transport/macfuse/mod.rs

```diff
@@ -0,0 +1,1177 @@
+// Copyright (C) 2020 Alibaba Cloud. All rights reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+//! Traits and Structs to implement the /dev/fuse Fuse transport layer.
+
+use std::collections::VecDeque;
+use std::fmt;
+use std::io::{self, IoSlice, Write};
+use std::marker::PhantomData;
+use std::mem::ManuallyDrop;
+use std::os::unix::io::RawFd;
+
+use nix::sys::uio::{writev, IoVec};
+use vm_memory::{ByteValued, VolatileMemory, VolatileMemoryError, VolatileSlice};
+
+use super::{FileReadWriteVolatile, IoBuffers, Reader};
+use crate::BitmapSlice;
+
+mod session;
+mod macfuse;
+pub use session::*;
+use crate::transport::FileVolatileSlice;
+
+/// Error codes for Virtio queue related operations.
+#[derive(Debug)]
+pub enum Error {
```

Comment:
maybe it's better to extract this Error enum into a separate mod, that can be referenced by both macfuse and fusedev feature